canonical-web-and-design/canonical.com
# Reduce parsing of department fields

## Details

- Follow up to https://github.com/canonical-web-and-design/canonical.com/pull/273; don't review or merge this before that one is merged.
- It was previously discussed that it would be good to minimise the use of logic and field mapping in middleware and try to use data straight from the Harvest API as much as possible.<details>![image](https://user-images.githubusercontent.com/52562977/90923207-28d0a600-e3e5-11ea-9e9a-f4e9e6ab2a58.png)</details>
- I apologise for the format and timing of this PR; I would have like to wait for #273 to be merged before pushing this up, but as I am leaving the company today I thought it might be worth pushing this up so you have the option to take it or leave it as you wish.

## Done

- Removed the `__parse_feed_department` function from the `Department` class in `app.py`
- Reduced the number of fields in the `_parse_feed_department` function in `greenhouse.py`
- Renamed several files in `/templates/careers` to match the department names on the Harvest API

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8002/careers/all
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Ensure that all department fields are present in the navigation (I believe there should be 12 at this time)
- Ensure that the vacancy counters in the navigation, as well as the number of vacancies that actually show up, are still accurate
- Ensure that all departments that have corresponding templates actually render properly instead of the generic placeholder (I believe this should be all departments except for `/careers/product`)
